src/main/msp/msp_serial.c

```diff
@@ -290,7 +292,13 @@ static int mspSerialSendFrame(mspPort_t *msp, const uint8_t * hdr, int hdrLen, c
     //  b) Response fits into TX buffer
     const int totalFrameLength = hdrLen + dataLen + crcLen;
     if (!isSerialTransmitBufferEmpty(msp->port) && ((int)serialTxBytesFree(msp->port) < totalFrameLength)) {
+#ifdef USE_MSP_DISPLAYPORT
+        //Currently TxBuffer free space is not enough, Wait for TxBuffer empty
+        const int perByteCostUs = (1000000 / (msp->port->baudRate / 10));
+        delayMicroseconds((int)serialTxBytesFree(msp->port) * perByteCostUs);
```

Comment:
No, as you were told before, busy waiting in here is **definitely** not acceptable. Please start listening to instructions you are given.

Reply:
I deleted this part and modified UART_TX_BUFFER_SIZE to a suitable value